var/spack/repos/builtin/packages/autoconf/package.py

```diff
@@ -35,7 +35,9 @@ class Autoconf(Package):
     version('2.69', '82d05e03b93e45f5a39b828dc9c6c29b')
     version('2.62', '6c1f3b3734999035d77da5024aab4fbd')
 
-    depends_on('m4', type='build')
+    # Note: m4 is not a build-time dependency of autoconf. m4 is
+    # needed when autoconf runs, not when autoconf is built.
+    depends_on('m4')
```

Comment:
@eschnett my comment is based on what you added to the pr. If it is needed during build + run, then it should be `build` + `run`. 

What you have now is ("build", "link"), so it does not solve the problem.